src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java

```diff
@@ -0,0 +1,231 @@
+import uk.gov.hmcts.reform.civil.service.flowstate.IStateFlowEngine;
+import uk.gov.hmcts.reform.civil.service.referencedata.LocationReferenceDataService;
+import uk.gov.hmcts.reform.civil.utils.CourtLocationUtils;
+import uk.gov.hmcts.reform.idam.client.models.UserInfo;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Optional.ofNullable;
+import static uk.gov.hmcts.reform.civil.callback.CallbackParams.Params.BEARER_TOKEN;
+import static uk.gov.hmcts.reform.civil.enums.CaseRole.RESPONDENTSOLICITORONE;
+import static uk.gov.hmcts.reform.civil.enums.CaseRole.RESPONDENTSOLICITORTWO;
+import static uk.gov.hmcts.reform.civil.enums.YesOrNo.NO;
+import static uk.gov.hmcts.reform.civil.enums.YesOrNo.YES;
+import static uk.gov.hmcts.reform.civil.service.flowstate.FlowFlag.TWO_RESPONDENT_REPRESENTATIVES;
+
+@Component
+public class PopulateRespondentCopyObjects implements CaseTask {
+
+    private final ObjectMapper objectMapper;
+    private final CourtLocationUtils courtLocationUtils;
+    public static final String ERROR_DEFENDANT_RESPONSE_SUBMITTED =
+        "There is a problem"
+            + "\n"
+            + "You have already submitted the defendant's response";
+    private final LocationReferenceDataService locationReferenceDataService;
+    private final IStateFlowEngine stateFlowEngine;
+    private final CoreCaseUserService coreCaseUserService;
+    private final UserService userService;
+    private final LocationReferenceDataService locationRefDataService;
+
+    public PopulateRespondentCopyObjects(ObjectMapper objectMapper,
+                                         CourtLocationUtils courtLocationUtils,
+                                         LocationReferenceDataService locationReferenceDataService,
+                                         IStateFlowEngine stateFlowEngine,
+                                         CoreCaseUserService coreCaseUserService,
+                                         UserService userService,
+                                         LocationReferenceDataService locationRefDataService) {
+
+        this.objectMapper = objectMapper;
+        this.courtLocationUtils = courtLocationUtils;
+        this.locationReferenceDataService = locationReferenceDataService;
+        this.stateFlowEngine = stateFlowEngine;
+        this.coreCaseUserService = coreCaseUserService;
+        this.userService = userService;
+        this.locationRefDataService = locationRefDataService;
+    }
+
+    @Override
+    public CallbackResponse execute(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        LocalDateTime dateTime = LocalDateTime.now();
+
+        CallbackResponse errorResponse = checkSubmitError(callbackParams, caseData, dateTime);
+        if (errorResponse != null) {
+            return errorResponse;
+        }
+
+        var isRespondent1 = findOutIsRespondent1(callbackParams);
```

Comment:
I would try not using var, I know the latest java interactions allow this but I believe the strong typification in Java is on of its best features so I'd stick to strongly typing variables.